lib/netext/grpcext/conn.go

```diff
@@ -0,0 +1,313 @@
+// Package grpcext allows gRPC requests collecting stats info.
+package grpcext
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"go.k6.io/k6/js/modules"
+	"go.k6.io/k6/metrics"
+
+	protov1 "github.com/golang/protobuf/proto" //nolint:staticcheck,nolintlint // this is the old v1 version
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	grpcstats "google.golang.org/grpc/stats"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/encoding/prototext"
+	"google.golang.org/protobuf/proto"
```

Comment:
Do you consider using `github.com/gogo/protobuf/proto` which has faster mashalling/unmarshalling?

Reply:
Hi @cygnrh, thanks for your review.

Unfortunately, we would prefer to not include it, we are very meticulous in adding new dependencies to `k6`, you can read some motivations in [Dependencies.md](https://github.com/grafana/k6/blob/master/Dependencies.md).
We would prefer to fix allowing the marshaler and unmarshaler to be injected, it would allow us to remove also the tightly coupled JSON dependency. I opened an issue for it https://github.com/grafana/k6/issues/2497

If you have relevant observations for the grpc module, it would be helpful to get your feedback in a new issue or in  https://github.com/grafana/k6/issues/1846.

Reply:
We haven't observed performance issue since we only did the initial test. If we do see anything as k6 gRPC being more widely adopted, I'll post in the thread https://github.com/grafana/k6/issues/1846.